pageserver/src/page_cache.rs

```diff
@@ -816,6 +822,8 @@ impl PageCache {
     fn new(num_pages: usize) -> Self {
         assert!(num_pages > 0, "page cache size must be > 0");
 
+        // We use Box::leak here and into_boxed_slice to avoid leaking uninitialized
+        // memory that Vec's might contain.
```

Comment:
Not sure this comment is accurate. The point is to create a large array to back the page cache, with static lifetime. I don't think we're worried about uninitialized memory.

Instead of a static lifetime, a more modular way to do this would be for PageCache to have a lifetime, and this array and `SlotInner.buf` would have the same lifetime. Currently, they are all just 'static, which is fine as we only have one page cache in the process. So I'm not suggesting to change any code here, just explaining why this needs to be 'static..